test/e2e/tests/test_ca_hierarchy.py

```diff
@@ -0,0 +1,269 @@
+    )
+
+    suffix = random_suffix_name("", 10)
+    replacements["NAME"] = sub_ca_name
+    replacements["COMMON_NAME"] = "www.example" + suffix + ".com"
+    replacements["COUNTRY"] = "US"
+    replacements["LOCALITY"] = "Arlington"
+    replacements["ORG"] = "Example Organization " + suffix
+    replacements["STATE"] = "Virginia"
+
+    # Load CA CR
+    sub_ca_resource_data = load_acmpca_resource(
+        "subordinate_certificate_authority",
+        additional_replacements=replacements,
+    )
+
+    # Create CA resource
+    sub_ca_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, "certificateauthorities",
+        sub_ca_name, namespace="default",
+    )
+
+    replacements["NAME"] = sub_cert_name
+    replacements["CA_NAME"] = ca_name
+    replacements["CSR_CA_NAME"] = sub_ca_name
+    replacements["CERTIFICATE_SEC_NS"] = sub_secret.ns
+    replacements["CERTIFICATE_SEC_NAME"] = sub_secret.name
+    replacements["CERTIFICATE_SEC_KEY"] = sub_secret.key
+    replacements["TEMPLATE_ARN"] = "arn:aws:acm-pca:::template/SubordinateCACertificate_PathLen2/V1"
+
+    # Load Certificate CR
+    sub_cert_resource_data = load_acmpca_resource(
+        "subordinate_certificate_ref",
+        additional_replacements=replacements,
+    )
+
+    # Create Certificate resource
+    sub_cert_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, "certificates",
+        sub_cert_name, namespace="default",
+    )
+
+    replacements["NAME"] = sub_act_name
+    replacements["CA_NAME"] = sub_ca_name
+    replacements["CERTIFICATE_CHAIN_SEC_NS"] = certificate_chain_secret.ns
+    replacements["CERTIFICATE_CHAIN_SEC_NAME"] = certificate_chain_secret.name
+    replacements["CERTIFICATE_CHAIN_SEC_KEY"] = certificate_chain_secret.key
+    replacements["CERTIFICATE_SEC_NS"] = sub_secret.ns
+    replacements["CERTIFICATE_SEC_NAME"] = sub_secret.name
+    replacements["CERTIFICATE_SEC_KEY"] = sub_secret.key
+    replacements["COMPLETE_CERTIFICATE_CHAIN_SEC_NS"] = sub_certificate_chain_secret.ns
+    replacements["COMPLETE_CERTIFICATE_CHAIN_SEC_NAME"] = sub_certificate_chain_secret.name
+    replacements["COMPLETE_CERTIFICATE_CHAIN_SEC_KEY"] = sub_certificate_chain_secret.key
+    
+    # Load CAActivation CR
+    sub_act_resource_data = load_acmpca_resource(
+        "subordinate_certificate_authority_activation_ref",
+        additional_replacements=replacements,
+    )
+    
+    # Create CAActivation resource
+    sub_act_ref = k8s.create_reference(
+        CRD_GROUP, CRD_VERSION, "certificateauthorityactivations",
+        sub_act_name, namespace="default",
+    )
+
+    k8s.create_custom_resource(ca_ref, ca_resource_data)
+    k8s.create_custom_resource(cert_ref, cert_resource_data)
+    k8s.create_custom_resource(act_ref, act_resource_data)
+    k8s.create_custom_resource(sub_ca_ref, sub_ca_resource_data)
+    k8s.create_custom_resource(sub_cert_ref, sub_cert_resource_data)
+    k8s.create_custom_resource(sub_act_ref, sub_act_resource_data)
```

Comment:
Rather than submitting these all seperately, can't we submit it all as part of a single template?